src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/GaugeMetric.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+import java.util.function.Supplier;
+
+public class GaugeMetric<T> extends Metric<T> {
+
+    private Supplier<T> LOAD_VALUE;
```

Comment:
why is the field name in all caps?

Reply:
I consider it as a function, so use all caps.

Reply:
Do we have any other code where we use all caps for function names? From what I've seen and used so far, only the names of constants should be in all caps.

Reply:
Not sure. I can change it to loadValue, is that OK for you?

Reply:
That is fine, thank you.